### PR TITLE
Make type parameter names in dataflow framework consistent.

### PR DESCRIPTION
**Pull request by xingweitian**
Currently, we both use `A extends AbstractValue<A>` and `V extends AbstractValue<V>` in dataflow framework. This PR changes all the usages of `A` to `V`.

### CODE REVIEW

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java [hunk 1]
```diff
@@ -400,13 +400,17 @@ protected void propagateStoresTo(
     /**
      * Updates the value of node {@code node} to the value of the {@code transferResult}. Returns
      * true if the node's value changed, or a store was updated.
+     *
+     * @param node a node
+     * @param transferResult the result type of a transfer function
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java:405]: `the transfer result to use` maybe? The `type` is confusing, b/c we use the value from that transfer result.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java [hunk 2]
```diff
@@ -416,8 +420,12 @@ protected boolean updateNodeValues(Node node, TransferResult<A, S> transferResul
 
     /**
      * Call the transfer function for node {@code node}, and set that node as current node first.
+     *
+     * @param node a node
+     * @param store the input type of a transfer function
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java:425]: Similarly, the `type` here doesn't seem necessary.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java [hunk 3]
```diff
@@ -678,16 +686,20 @@ public String toString() {
     /**
      * Read the {@link TransferInput} for a particular basic block (or {@code null} if none exists
      * yet).
+     *
+     * @param b a block {@code b}
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java:690]: the `{@code b}` part is unnecessary, here and below.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java [hunk 4]
```diff
@@ -743,24 +756,33 @@ public boolean isRunning() {
...
+     * @param in the current node values
+     */
+    /*package-private*/ void setNodeValues(IdentityHashMap<Node, V> in) {
         assert !isRunning;
         nodeValues.clear();
         nodeValues.putAll(in);
     }
 
     /**
+     * @param t a {@link Tree} t
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/Analysis.java:780]: Links to the types are unnecessary in the description - javadoc already links the types in the signature.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java [hunk 5]
```diff
@@ -21,12 +21,13 @@
  * providing the abstract values given a node or a tree. Note that it does not keep track of custom
  * results computed by some analysis.
  *
- * @param <A> type of the abstract value that is tracked
+ * @param <V> type of the abstract value that is tracked
+ * @param <S> the store type used in the analysis
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/AnalysisResult.java:25]: ConditionalTransferResult below uses
` * @param <S> the {@link Store} used to keep track of intermediate results`
Maybe also make all these descriptions consistent?
In this particular case, the two descriptions aren't symmetric, which makes them a bit ugly.

**Comment by wmdietl** [comment]
@xingweitian I think we had bad luck with timing... @mernst in #3350 is adding all missing javadoc, which overlaps with many of the changes you're making.
It might be better to wait for #3350 to be merged and then do the renaming.
@mernst What do you think about how to best resolve the conflicts between these two PRs?

**Comment by mernst** [comment]
@wmdietl @xingweitian There will be a few merge conflicts, but not terribly many.  I think I can resolve them if you wish.  Thanks for these improvements, by the way.

**Comment by xingweitian** [comment]
Let's wait until #3350 is merged. It is ok for me to finish this PR. Thanks!

**Comment by mernst** [comment]
@xingweitian #3350 is merged, is pulled into this PR's branch, and merge conflicts are resolved.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/TransferResult.java [hunk 6]
```diff
@@ -37,8 +38,18 @@
      *     produced
      * @param exceptionalStores the stores to use if the basic block throws an exception
      */
+    /**
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/TransferResult.java:41]: There already is some javadoc above. Can you merge the two?

**Comment by xingweitian** [comment]
@wmdietl Thanks for the review! I think I have solved the comments.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/ConditionalTransferResult.java [hunk 7]
```diff
@@ -32,31 +33,42 @@
...
      */
     public ConditionalTransferResult(
-            @Nullable A value, S thenStore, S elseStore, boolean storeChanged) {
+            @Nullable V value, S thenStore, S elseStore, boolean storeChanged) {
         this(value, thenStore, elseStore, null, storeChanged);
     }
 
     /**
-     * See {@link #ConditionalTransferResult(AbstractValue, Store, Store, Map, boolean)}.
+     * Create a new {@link #ConditionalTransferResult(AbstractValue, Store, Store)}.
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/ConditionalTransferResult.java:50]: Should the linked constructor be consistent with the `@see` constructor below? At the moment this links to itself, which doesn't make much sense.
The implementation actually calls a constructor without a map...
Should it be documented that `false` is used for the boolean?
Same comment for the constructor below this one.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/RegularTransferResult.java [hunk 8]
```diff
@@ -32,28 +33,36 @@
...
      * @see #RegularTransferResult(AbstractValue, Store, Map, boolean)
      */
-    public RegularTransferResult(@Nullable A value, S resultStore, boolean storeChanged) {
+    public RegularTransferResult(@Nullable V value, S resultStore, boolean storeChanged) {
         this(value, resultStore, null, storeChanged);
     }
 
     /**
-     * See {@link #RegularTransferResult(AbstractValue, Store, Map, boolean)}.
+     * Create a new {@link #RegularTransferResult(AbstractValue, Store)}.
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/RegularTransferResult.java:46]: Same comment about self-referential constructor docs in this class.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/RegularTransferResult.java [hunk 9]
```diff
@@ -32,28 +33,36 @@
      * class (including use through aliases). Complete control over the object is transfered to this
      * class.
      *
+     * @param value the abstract value produced by the transfer function
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/RegularTransferResult.java:36]: Can you add a summary for this constructor? At the top it currently has `* *`.

File: dataflow/src/main/java/org/checkerframework/dataflow/analysis/TransferResult.java [hunk 10]
```diff
@@ -31,14 +32,17 @@
     protected final @Nullable Map<TypeMirror, S> exceptionalStores;
 
     /**
-     * Create a new TransferResult.
+     * Create a {@link #TransferResult(AbstractValue, Map)}, given {@link #resultValue} and {@link
```
Comment (wmdietl) [inline dataflow/src/main/java/org/checkerframework/dataflow/analysis/TransferResult.java:35]: Another link to the same constructor.

**Comment by wmdietl** [comment]
@mernst Did you want to have a look at this?

**Pull request merged**